firmware/ventilator-controller-stm32/Core/Inc/Pufferfish/Util/Endian.h

```diff
@@ -17,7 +17,7 @@ namespace Pufferfish::Util {
 constexpr void read_ntoh(const uint8_t *network_endian, int16_t &host_endian) noexcept {
   uint16_t network_endian_value =
       (set_byte<1, uint16_t>(network_endian[1]) + set_byte<0, uint16_t>(network_endian[0]));
-  host_endian = HAL::ntoh(network_endian_value);
+  host_endian = HAL::STM32::ntoh(network_endian_value);
```

Comment:
Hmm, files in `Util/Endian.h` shouldn't be using classes/functions from `HAL::STM32`. It looks like the renaming exposed an issue I hadn't realized. I can think of two possible solutions, though there's a trade-off between them:

1. We could convert the HAL endian functions into methods of a HAL::Interface::Endian class, and then we'll have a HAL::STM32::Endian subclass. Then we need to convert the `Util/Endian.h` functions into methods of a `Util::Endian` class, and we'll need to pass a `HAL::Interface::Endian` object into its constructor, and we'll need to pass a `Util::Endian` class into the constructor of anything which needs to call it. This would follow the pattern established with `HAL::Interface::Time`, and it would make explicit the dependency of any class on platform-specific endian code, but it would could be kind of annoying.
2. In `HAL/Interfaces`, we could have create an `Endian.h` with forward declarations of the endian-related functions. Then in `HAL/STM32` we would create an `Endian.cpp` with the function definitions. Then we are using the linker to specify the implementation of the endian functions; however, it may be harder to troubleshoot because it all depends on us configuring the linker to compile the cpp files for `Core/Src/Pufferfish/HAL/STM32`.

Right now I'm leaning towards option 2 because the code will be simpler. What do you think?